examples/qp2q/README.md

```diff
@@ -0,0 +1,92 @@
+
+## Pre-procesing data
+* Download AOL Search Logs data from [here](http://www.cim.mcgill.ca/~dudek/206/Logs/AOL-user-ct-collection/).
+* Extract files into `RAW_DATA_DIR` folder. This folder should contain 10 files with name template: ` user-ct-test-collection-**.txt`.
+* Run the following command to pre-process the data and to create train/test/dev splits (as used in this paper).
+    - ```bash 
+      cd examples/qp2q 
+      python utils/process_aol_dataset_from_orig_aol_files.py --data_dir <RAW_DATA_DIR> --out_dir <DATA_DIR> 
+      ```
+* The processed data is present in `DATA_DIR` folder, which contains three folders: `train`, `test`, and `dev`.
+Each folder contains a `.json` file. Each line in the file contains a data point stored in JSON format.
+  
+
+## Training Models
+
+* First run the setup script in `qp2q/bin` folder
+    - ```bash 
+        cd examples/qp2q
+        source bin/setup.sh 
+      ```
+* Train a model
+    - ```python models/train_model.py --config <path/to/config/file>  ```
+    - Config files for various configurations of the models from [Table 2](https://arxiv.org/abs/2012.07654) in the paper 
+      are present in `qp2q/config` folder. 
+    - Please configure `fdir` parameter in config files to point to directory containing training data. By default,
+    training data is assumed to present in `data/aol/train` directory where `data` directory is assumed to be present 
+      in the same folder as `qp2q` folder.
+    - Trained models will be stored in folders under `results` folder.
+    - The `data`, `results` and `qp2q` folders are assumed to be present in the same directory, with the data 
+      directory organized as illustrated below.
+    ```
+    ├── qp2q
+    ├── results
+    ├── data
+    |    ├── aol
+    |        ├── train
+    |           ├── train.json
+    |        ├── test
+    |           ├── test.json
+    |        ├── val
+    |           ├── val.json
+    ```
+
+
+## Evaluation
+
+* To generate query suggestions and evaluate the predictions for proposed models: 
+    -  ```bash
+       python eval/run_eval.py --gt <path/to/gt/file> --out_dir <path/to/result/dir> --model_dir <path/to/trained_model/folder>
+       ```
+* To generate query suggestions and evaluate the predictions for Most-Frequent-Query (MFQ) baseline:
+    - First generate a dictionary mapping each prefix to a list of top-k (k=10) query suggestions and then run eval.  
+      ```bash 
+      python utils/create_pref_to_top_k_suggestions_dict.py --k 10 --fdir <train/data/folder> --out_file </output/filename>
+      python eval/run_mfq_eval.py --gt <path/to/gt/file> --out_dir <path/to/result/dir> --topk_file <path/to/prefix/to/topk/query/file> 
+      ```
+  
+  
+## Citation
+
+If you find the code useful, please consider citing our paper.
+
+* [Session-Aware Query Auto-completion using Extreme Multi-label Ranking (Yadav et al., KDD 2021)](https://arxiv.org/pdf/2012.07654.pdf)  [[bib]](../../bibtex/yadav2021session.bib)
+
```

Comment:
I realized the following bibtex content is not consistent with the `../../bibtex/yadav2021session.bib`. For example:
(1) `@article` vs `@ inproceedings `
(2) `journal` vs `booktitle `
Please consider making both consistent.

Reply:
Updated!